File: isi-backend/src/main/java/com/guuri11/isi/infrastructure/persistance/CommandRepository.java
package com.guuri11.isi.infrastructure.persistance;

import com.guuri11.isi.domain.chat.Chat;
import com.guuri11.isi.domain.command.Command;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

public interface CommandRepository extends JpaRepository<Command, UUID> {
    List<Command> findByChat(Chat chat);
}

File: lib/lib/src/main/kotlin/org/isi/domain/repository/CommandRepository.kt
package org.isi.domain.repository

import org.isi.domain.models.Chat
import org.isi.domain.models.Command
import org.isi.domain.models.TaskType

interface CommandRepository {
    suspend fun findAll(): List<Command>
    suspend fun create(messages: List<Command>, chat: Chat?, task: TaskType?, apiKey: String?): Command
}